backend/apps/rag/utils.py

```diff
@@ -236,6 +237,21 @@ def generate_multiple(query, f):
         return lambda query: generate_multiple(query, func)
 
 
+def _extract_collection_names(file):
+    if file["type"] == "collection":
+        # keep for backward compatibility
+        return file["collection_names"]
+    elif file["type"] == "all_documents":
+        return [document.collection_name for document in Documents.get_docs()]
+    elif file["type"] == "tag":
+        return [
+            document.collection_name
+            for document in Documents.get_docs_by_tag(file["name"])
+        ]
+    else:
+        return [file["collection_name"]]
```

Comment:
There's no need to have `elif` here since all your branches are returning something.

```suggestion
    if file["type"] == "collection":
        # keep for backward compatibility
        return file["collection_names"]
    
    if file["type"] == "all_documents":
        return [document.collection_name for document in Documents.get_docs()]
    
    if file["type"] == "tag":
        return [
            document.collection_name
            for document in Documents.get_docs_by_tag(file["name"])
        ]

    return [file["collection_name"]]
```